plugin/kprom/kprom.go

```diff
@@ -130,9 +131,19 @@ func (m *Metrics) OnNewClient(client *kgo.Client) {
 		subsystem   = m.cfg.subsystem
 		constLabels prometheus.Labels
 	)
+	dynamicLabels := []string{"node_id"}
 	if m.cfg.withClientLabel {
 		constLabels = make(prometheus.Labels)
 		constLabels["client_id"] = client.OptValue(kgo.ClientID).(string)
+		dynamicLabels = append(dynamicLabels, "client_id")
+		if !slices.Contains(m.cfg.fetchProduceOpts.labels, "client_id") {
+			m.cfg.fetchProduceOpts.labels = append(m.cfg.fetchProduceOpts.labels, "client_id")
+		}
+	}
+
+	// Skip metrics definitions if hook was already called
+	if m.connConnectsTotal != nil {
```

Comment:
This isn't concurrent safe -- do you intend to initialize clients concurrently?

Reply:
For what it's worth, if so, the logic can be extracted to a guarded initialization (and other hooks can have a quick check to see if initialization has completed). Let me know if you'd like me to add that.